sympy/physics/mechanics/tests/test_kane.py

```diff
@@ -9,6 +9,20 @@
 from sympy.core.backend import USE_SYMENGINE
 
 
+def test_invalid_coordinates():
+    # Simple pendulum, but use symbols instead of dynamicsymbols
+    l, m, g = symbols('l m g')
+    q, u = symbols('q u')  # Generalized coordinate
+    kd = [q.diff(dynamicsymbols._t) - u]
+    N, O = ReferenceFrame('N'), Point('O')
+    O.set_vel(N, 0)
+    P = Particle('P', Point('P'), m)
+    P.point.set_pos(O, l * (sin(q) * N.x - cos(q) * N.y))
+    F = (P.point, -m * g * N.y)
+    raises(ValueError, lambda: KanesMethod(N, [q], [u], kd, bodies=[P],
+                                           forcelist=[F]))
```

Comment:
This test would also pass, when the `_validate_coordinates()` call in `KanesMethod.__init__` would be removed (only the error message is now a bit more clear).